src/android/Device.java

```diff
@@ -59,31 +66,105 @@ public void initialize(CordovaInterface cordova, CordovaWebView webView) {
+     * pluginResult will be modified and returned in one of the callbacks
+     * of "retrieveZebraSerialNumber()"
+     *
+     * In any other cases or devices, pluginResult is returned immediately.
+     *
+     * @param callbackContext   The callback id used when calling back into JavaScript.
+     * @return
+     */
+    private void getDeviceInfo(CallbackContext callbackContext) throws JSONException {
+        JSONObject pluginResult = new JSONObject();
+        pluginResult.put("uuid", Device.uuid);
+        pluginResult.put("version", this.getOSVersion());
+        pluginResult.put("platform", this.getPlatform());
+        pluginResult.put("model", this.getModel());
+        pluginResult.put("manufacturer", this.getManufacturer());
+        pluginResult.put("isVirtual", this.isVirtual());
+
+        // Due to privacy restrictions ALL devices running on Android 10 or higher
+        // "getSerialNumber()" will just return the string "unknown" instead of the real device serial number ...
+        pluginResult.put("serial", this.getSerialNumber());
+
+        // ... luckily, at least zebra devices offer a way to deliver their serial number on Android 10 or higher.
+        // So in case the device is a zebra device running on Android >= 10
+        // we are trying to (asynchronously) access the serial number via a DeviceIdentifiersWrapper
+        // and replace the "unknown" serial number in the PluginResult with the real serial number.
+        if (this.isZebraDevice() && android.os.Build.VERSION.SDK_INT >= 29) {
+            this.retrieveZebraSerialNumber(this.cordova.getContext(), callbackContext, pluginResult);
+        } else {
+            // in any other case, we instantly return the collected device values back to the caller
+            // (serial number will be "unknown" if device runs Android 10 or higher)
+            callbackContext.success(pluginResult);
+        }
+    }
+
+    /**
+     * Tries to retrieve the serial number of a Zebra device using DeviceIdentifiersWrapper.
+     *
+     * For more information see:
+     * https://github.com/ZebraDevs/DeviceIdentifiersWrapper
+     *
+     * @param context            The Android context
+     * @param callbackContext    The callback id used when calling back into JavaScript.
+     * @param pluginResult       A JSONObject which contains already gathered device info (like OS, platform, model, serial etc...)
+     */
+    private void retrieveZebraSerialNumber(Context context, CallbackContext callbackContext, JSONObject pluginResult)
+    {
+        DIHelper.getSerialNumber(context, new IDIResultCallbacks() {
+            @Override
+            public void onSuccess(String serialNumber)  {
+                try {
+                    // replace "unknown" with the real serial number in the pluginResult
+                    pluginResult.put("serial", serialNumber);
+                } catch (JSONException e) {
+                    Log.i(TAG, e.getMessage());
```

Comment:
Should be log level error and should include information about context:

```suggestion
                    Log.e(TAG, "error while assigning serial number to JSONObject: " + e.getMessage());
```